README.md

```diff
@@ -1,57 +1,269 @@
+
+#### closedClassName
+
+Type: `string` | Required: `false`
+
+A className to apply to the accordion container when all the items are closed.
+
+#### itemContainerClassName
+
+Type: `string` | Required: `false`
+
+A className to apply for each accordion item's container.
+
+#### headingContainerClassName
+
+Type: `string` | Required: `false`
+
+A className to apply to the heading container of each item.
+
+#### titleContainerClassName
 
-#### {package-api-prop-example}
+Type: `string` | Required: `false`
 
-Type: `object`
-Required: `true`
+A className to apply to the title of each item.
+
+#### contentContainerClassName
+
+Type: `string` | Required: `false`
+
+A className to apply to the content container of each item.
+
+#### contentClassName
+
+Type: `string` | Required: `false`
+
+A className to apply to the content of each item.
+
+#### iconClassName
+
+Type: `string` | Required: `false`
+
+A className to apply to the icon container of each item.
+
+#### onItemToggling
+
+Type: `function` | Required: `false` | Arguments: `index: Number, isActive: Object`
+
+Invoked before an item is opened or closed.
+
+#### onItemToggle
+
+Type: `function` | Required: `false` | Arguments: `index: Number, isActive: Object`
+
+Invoked after an item is opened or closed.
+
+#### scrollIntoView
+
+Type: `boolean` | Required: `false` | Default: `false`
+
+Whether or not the content of an item should be scrolled to be visible.
+
+### AccordionItem Props
+
+#### title
+
+Type: `any` | Required: `true`
+
+The content of the title.
+
+#### children
+
+Type: `node` or `function` | Required: `false`
+
+The content of the item.  
+Example with function:
 
-The `{package-api-prop-example}` has the following shape:
 ```js
-{package-api-prop-example}: PropTypes.shape({
-    foo: PropTypes.string,
-    bar: PropTypes.arrayOf(PropTypes.object),
-}).isRequired,
+<AccordionItem { ...props }>
+    { ({ isActive }) => isActive ? 'Open' : 'Closed' }
+</AccordionItem>
+```
+
+#### animationDuration
+
+Type: `number` | Required: `false` | Default: `300`
+
+Duration of the animation when the item is opened or closed.
+
+#### animationEasing
+
+Type: `string` | Required: `false` | Default: `ease-in-out`
+
+Type of animation easing when the item is opened or closed.
+
+#### itemContainerClassName
+
+Type: `string` | Required: `false`
+
+A className to apply to the item container.
+
+#### headingContainerClassName
+
+Type: `string` | Required: `false`
+
+A className to apply to the heading container.
+
+#### titleContainerClassName
+
+Type: `string` | Required: `false`
+
+A className to apply to the title.
+
+#### contentContainerClassName
+
+Type: `string` | Required: `false`
+
+A className to apply to the content container.
+
+#### contentClassName
+
+Type: `string` | Required: `false`
+
+A className to apply to the content.
+
+#### iconClassName
+
+Type: `string` | Required: `false`
+
+A className to apply to the icon container.
+
+#### icon
```

Comment:
Same for the above props. We can define all of these props for the Accordion and for AccordionItems.

Is the expected behaviour here to use the Accordion props as default ones and if there is a collision we use the prop provided to the AccordionItem? Is there an override or an augmentation?

Reply:
https://github.com/moxystudio/react-accordion/pull/1#discussion_r417357711